diff --git a/MdePkg/Include/Base.h b/MdePkg/Include/Base.h
index 4680e64136..f6fac7eb42 100644
--- a/MdePkg/Include/Base.h
+++ b/MdePkg/Include/Base.h
@@ -801,7 +801,7 @@ typedef UINTN  *BASE_LIST;
 **/
 #ifdef MDE_CPU_EBC
   #define STATIC_ASSERT(Expression, Message)
-#elif _MSC_EXTENSIONS
+#elif _MSC_VER
   #define STATIC_ASSERT static_assert
 #else
   #define STATIC_ASSERT _Static_assert
