--- dnsmasq.conf.default	2015-08-05 17:23:53.000000000 -0700
+++ dnsmasq.conf	2017-04-05 15:47:17.579607425 -0700
@@ -8,6 +8,7 @@
 # (53). Setting this to zero completely disables DNS function,
 # leaving only DHCP and/or TFTP.
 #port=5353
+port=0
 
 # The following two options make you a better netizen, since they
 # tell dnsmasq to filter out queries which the public DNS cannot
@@ -92,6 +93,9 @@
 # interface (eg eth0) here.
 # Repeat the line for more than one interface.
 #interface=
+interface=p3p1
+interface=p3p2
+
 # Or you can specify which interface _not_ to listen on
 #except-interface=
 # Or which to listen on by address (remember to include 127.0.0.1 if
@@ -109,7 +113,7 @@
 # want dnsmasq to really bind only the interfaces it is listening on,
 # uncomment this option. About the only time you may need this is when
 # running another nameserver on the same machine.
-#bind-interfaces
+bind-interfaces
 
 # If you don't want dnsmasq to read /etc/hosts, uncomment the
 # following line.
@@ -143,6 +147,10 @@
 # repeat this for each network on which you want to supply DHCP
 # service.
 #dhcp-range=192.168.0.50,192.168.0.150,12h
+dhcp-range=10.69.0.200,proxy
+dhcp-range=10.68.0.200,proxy
+
+dhcp-no-override
 
 # This is an example of a DHCP range where the netmask is given. This
 # is needed for networks we reach the dnsmasq DHCP server via a relay
@@ -424,6 +432,10 @@
 #dhcp-boot=tag:!gpxe,undionly.kpxe
 #dhcp-boot=mybootimage
 
+#dhcp-boot=tag:BC_EFI,BOOTX64.EFI
+dhcp-boot=BOOTX64.EFI,"dnsmasq_server",10.69.0.10
+#dhcp-boot=pxelinux.0,"dnsmasq_server",10.69.0.10
+
 # Encapsulated options for Etherboot gPXE. All the options are
 # encapsulated within option 175
 #dhcp-option=encap:175, 1, 5b         # priority code
@@ -449,6 +461,9 @@
 # Available boot services. for PXE.
 #pxe-service=x86PC, "Boot from local disk"
 
+pxe-service=BC_EFI,"Dnsmasq PXE boot",BOOTX64.EFI,10.69.0.10
+#pxe-service=BC_EFI,"Dnsmasq PXE boot",pxelinux.0,10.69.0.10
+
 # Loads <tftp-root>/pxelinux.0 from dnsmasq TFTP server.
 #pxe-service=x86PC, "Install Linux", pxelinux
 
@@ -623,7 +638,7 @@
 #log-queries
 
 # Log lots of extra information about DHCP transactions.
-#log-dhcp
+log-dhcp
 
 # Include another lot of configuration options.
 #conf-file=/etc/dnsmasq.more.conf
